spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -83,9 +88,11 @@
 
 	private final Function<ProducerRecord<?, ?>, KafkaOperations<?, ?>> templateResolver;
 
+	private final BitSet whichHeaders = new BitSet(10);
```

Comment:
Hm. They say that a new array is allocated for every `values()` call, so I don't know whether it would make sense to also cache the length value...

One more question. What's the motivation for not using `EnumSet<HeaderNames.HeadersToAdd>` instead of `BitSet`? The [documentation](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/EnumSet.html) says that:

>  Enum sets are represented internally as bit vectors.

Reply:
After reading `EnumSet` JavaDocs, I concur that this might be a better way for our use-case.

Reply:
Agreed; I wasn't aware of it.